zone/bot.cpp

```
@@ -4977,23 +4979,23 @@ void Bot::Damage(Mob *from, int64 damage, uint16 spell_id, EQ::skills::SkillType
+				parse->EventBot(EVENT_ATTACK, this, from, "", 0);
+			}
 		}
-	}
 
-	attacked_timer.Start(CombatEventTimer_expire);
+		attacked_timer.Start(CombatEventTimer_expire);
+	}
 	// if spell is lifetap add hp to the caster
-	if (IsValidSpell(spell_id) && IsLifetapSpell(spell_id)) {
+	if (from && IsValidSpell(spell_id) && IsLifetapSpell(spell_id)) {
 		int64 healed = GetActSpellHealing(spell_id, damage);
```

Clang (14.0.6)

[warning] variable 'healed' is not initialized

(cppcoreguidelines-init-variables)